scripts/lambdaRun.py

```diff
@@ -2,27 +2,23 @@
 import sys
 import os
 import re
-import yaml
-import json
-
-from collections import ChainMap
 
 from helpers.logHelpers import createLog
 from helpers.errorHelpers import InvalidExecutionType
-from helpers.configHelpers import loadEnvFile
-from helpers.clientHelpers import createAWSClient
+from helpers.clientHelpers import createEventMapping
+from helpers.configHelpers import setEnvVars
 
 logger = createLog('runScripts')
 
-# This script is invoked by the Makefile in root to execute various
-# commands around a python lambda. This includes deployment, local invocations,
-# and tests/test coverage. It attempts to replicate some of the functionality
-# provided through node/package.json
-# H/T to Paul Beaudoin for the inspiration
-
 
 def main():
+    """Invoked by the makefile's arguments, controls the overall execution of
+    the Lambda function. h/t to nonword for inspiration to use a makefile.
 
+    Raises:
+        InvalidExecutionType: If the args do not contain a valid execution type
+        raise an error.
+    """
     if len(sys.argv) != 2:
```

Comment:
Is it more important that there is at least one? E.g.:

    try:
        runType = sys.argv[1]
    except IndexError:
        logger.warning('Missing argument')
        sys.exit(1)

Reply:
likely yes? I'm unsure what's the best practice here. Additional `args` will have no effect, but should that be a valid execution type? I'm torn between being exacting in how this is invoked and tolerating additional parameters that will have no effect either way. Perhaps it should raise an `info` message if more than one arg is supplied and still execute?

Reply:
So what could happen? Maybe through a copy-paste error you end up with something like

    scripts.lambdaRun development build-development

Where you mean for the second thing to run. If that is potentially really bad, and/or hard to track down, maybe it _is_ better to check that there is only one argument?

Reply:
The worst thing that could happen is that you accidentally deploy or deploy to the wrong environment. Of course you can do that anyway, but if you hit `make deploy production` when you wanted `make deploy development` I'm not sure there's much we can do to help. (Perhaps a future feature should prompt the user to confirm production deployments, or supply a flag to not require that).

Hypothetically local testing could be destructive too, but if your local code is integrated to a feature in production, again, not much we can do.

My idea was that more than one argument is not intended and should be flagged because we have no way of knowing what the intended command was. 

Also, perhaps I'm not thinking this through, but I don't think this should ever be directly invoked by a user, it should only handle calls from the `Makefile`, though I guess we should support people building different commands that they want to use?

Reply:
Maybe just leave it as is.